ystia/yorc
# Feature/deployment update

# Pull Request description

## Description of the change

Updated REST API and CLI documentation for the interface allowing now to update a deployment in premium version, while this operation is still not allowed in the open source version.

### What I did

Updated doc at `rest/http_api.md` and `doc/cli.rst` to describe that on a Yorc premium version a deployment update is performed sending a `PUT /deployments/<deployment_id>` with an existing deployment ID, or using CLI `yorc deployments deploy --id <deployment_id>` with an existing deployment ID. While the Open Source version of Yorc still fails as the deployment updated in only supported in the premium version.

Other changes in the code don't change the functionalities, but are just implemented to prepare the update operation inplementation in the premium version:
  * introducing new deployment statuses in `deployments/structs.go` related to updates: `UPDATE_IN_PROGRESS`,  `UPDATED`, `UPDATE_FAILURE`
  * adding test data and a stub test `deployments/update_store_oss_test.go` compiled only in the open source version thanks to build tag `!premium`
  * refactoring code in  `rest/deployments.go` to call new function`updateDeployment()` implemented in  `rest/updates_oss.go`, compiled only in the open source version, whose implemtation here fails on error reporting the update in supported in the premium version only.

### How to verify it

Install the full stack Alien4Cloud/Yorc using Yorc Alien4Cloud plugin from https://github.com/ystia/yorc-a4c-plugin/tree/feature/GH-112-deployment-update-workflows.

Deploy an application.

Once the Application is deployed, select this application in Alien4Cloud. In tab `Prepare next deployment`, click on `Update`.
Check an error occurs asking to have a look at error details in logs.
Check yorc logs show an error describing the update operation is only supported in premium version


### Description for the changelog

REST API doc changes on deployment update support in premium version ([GH-352](https://github.com/ystia/yorc/issues/352))

## Applicable Issues

https://github.com/ystia/yorc/issues/352
